Expand CodeQL coverage to all JavaScript and TypeScript

jonkoops opened the pull request:
Closes #32987

<!---
Please read https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

jonkoops committed:
Expand CodeQL coverage to all JavaScript and TypeScript

Closes #32987

Signed-off-by: Jon Koops <jonkoops@gmail.com>

stianst reviewed (commented):
Why are you removing themes? That's fully of HTML files, and surely should remain in the scan?

jonkoops commented:
> Why are you removing themes? That's fully of HTML files, and surely should remain in the scan?

I would not say it's 'full of HTML files', it's full of FreeMarker templates, [which CodeQL cannot scan](https://codeql.github.com/docs/codeql-overview/supported-languages-and-frameworks/). There are a couple of HTML files scattered around the project, but none of these exist within the themes directory.

I've gone ahead and added an additional trigger for `.html` files, and added some documentation to clarify this.

abstractj commented:
@jonkoops approving it after reviewing the chances and experimenting on my fork.
